### ISSUE TITLE
Doctor should expose a safe recovery path for stale JSONL merge artifacts

### ISSUE THREAD

**Issue by Dicklesworthstone** [Dicklesworthstone/beads_rust#344]
## Summary

`br doctor --json` can detect stale JSONL merge artifacts in `.beads/`, but the current robot contract does not provide a safe, machine-actionable recovery path for agents. In the live `eidetic_engine_cli` workspace this left automation in an avoidable gray zone: `br sync --status` reported DB/JSONL in sync, while `br doctor` still warned about `beads.left.jsonl` / `beads.base.jsonl` and claim gates treated tracker authority as degraded.

A human/agent eventually repaired it non-destructively by snapshotting `.beads/` and moving the obsolete merge artifacts into the timestamped recovery directory. After that, `br doctor` and `br sync --status` both reported clean. That workflow should be exposed by Beads itself, either as structured recovery actions in `doctor` or as an explicit repair command.

Related but distinct:

- #329 covers `doctor --no-db` not being DB-free.
- #330 covers `doctor.ok` / `sync.metadata` operational-authority ambiguity.
- #334 covers the shared Beads/BV claimability gate.

This issue is narrower: once `jsonl.merge_artifacts` is detected, Beads should tell a robot exactly how to inspect and quarantine stale merge artifacts without deleting data or guessing.

## Environment

```text
workspace: /Users/jemanuel/projects/eidetic_engine_cli
br version observed in related reports: br 0.1.13
observed: 2026-06-08
configured db: null, using canonical .beads/beads.db
records: 3612
```

## Observed pre-repair state

A live incident update on 2026-06-08T17:45Z recorded this combination:

```text
br sync --status --json:
  dirty_count=0
  jsonl_newer=false
  db_newer=false

br doctor --json:
  jsonl.merge_artifacts warned for beads.left.jsonl and beads.base.jsonl
  sync.metadata still used External changes pending import wording
```

The specific Beads comment from the local tracker recorded:

```text
br sync is clean (`dirty_count=0`, `jsonl_newer=false`, `db_newer=false`) but br doctor still reports merge artifacts (`beads.left.jsonl`, `beads.base.jsonl`) and `External changes pending import`, so tracker/claim authority is still degraded.
```

That state is hard for an agent to interpret safely:

- DB and JSONL content are in sync.
- JSONL parses and counts match.
- Merge artifacts still sit beside the canonical JSONL.
- `doctor` reports the artifact warning, but no structured next action says whether they are stale, active conflicts, or safe to quarantine.
- `sync --status` does not surface the artifact warning, so automation must reconcile multiple command surfaces.

## Manual recovery that worked

The repair used the safe pattern from the local `fixing-beads-problems` playbook:

1. Snapshot `.beads/` into a timestamped recovery directory.
2. Move obsolete merge artifacts into that recovery directory, not delete them.
3. Re-run `br doctor --json` and `br sync --status --json`.

After repair, current live checks report:

```json
{
  "doctor": {
    "ok": true,
    "jsonl.merge_artifacts": "ok",
    "jsonl.parse": "Parsed 3612 records",
    "counts.db_vs_jsonl": "Both have 3612 records",
    "sync.metadata": "Database and JSONL are in sync"
  },
  "sync_status": {
    "dirty_count": 0,
    "jsonl_newer": false,
    "db_newer": false,
    "jsonl_content_hash": "a3bbef64b5083d9da3a8585c5d5e119a685a7900959d60789379c2601e1afcc5"
  },
  "recovery_dir": ".beads/recovery_20260608T174247Z"
}
```

The successful local Beads comment says:

```text
Beads storage hygiene repaired non-destructively with fixing-beads-problems workflow. Snapshot is .beads/recovery_20260608T174247Z; obsolete beads.left/base JSONL+meta merge artifacts were moved into that recovery directory, not deleted. br doctor --json and --no-db now report jsonl.merge_artifacts ok, counts.db_vs_jsonl ok with 3612 records, sqlite.integrity_check ok, and sync.metadata 'Database and JSONL are in sync' with dirty_issues=0.
```

## Expected behavior

When `br doctor --json` sees JSONL merge artifacts, it should expose enough machine-readable recovery context for agents to avoid both unsafe writes and manual guesswork.

At minimum, `jsonl.merge_artifacts` should include structured fields like:

```json
{
  "name": "jsonl.merge_artifacts",
  "status": "warn",
  "message": "Merge artifacts detected in .beads/",
  "details": {
    "files": ["beads.left.jsonl", "beads.base.jsonl"],
    "canonical_jsonl": ".beads/issues.jsonl",
    "classification": "stale_or_unresolved_merge_artifacts",
    "safe_for_writes": false,
    "recovery": [
      {
        "kind": "snapshot",
        "command": "br repair merge-artifacts --snapshot --json"
      },
      {
        "kind": "inspect",
        "command": "br doctor --json"
      }
    ]
  }
}
```

Better would be an explicit non-destructive repair command, for example:

```bash
br repair merge-artifacts --snapshot --quarantine --json
```

Suggested response shape:

```json
{
  "ok": true,
  "changed": true,
  "snapshot_dir": ".beads/recovery_YYYYMMDDTHHMMSSZ",
  "moved": [
    {"from":".beads/beads.left.jsonl", "to":".beads/recovery_YYYYMMDDTHHMMSSZ/beads.left.jsonl"},
    {"from":".beads/beads.base.jsonl", "to":".beads/recovery_YYYYMMDDTHHMMSSZ/beads.base.jsonl"}
  ],
  "deleted": [],
  "postcheck": {
    "doctor_ok": true,
    "merge_artifacts_present": false,
    "dirty_count": 0,
    "jsonl_newer": false,
    "db_newer": false
  }
}
```

Important safety constraints:

- Never delete merge artifacts automatically.
- Always snapshot before quarantine.
- Refuse or require `--force` if artifacts contain conflict markers or if both DB and JSONL have unsynced value.
- Keep the command safe for agents in dirty multi-agent checkouts.
- Include postcheck results so a robot can know whether tracker authority improved.

## Why this matters

Agents currently have to infer Beads authority by combining `doctor`, `sync --status`, `show`, and local recovery playbooks. That is brittle during exactly the scenario where Beads is supposed to coordinate work safely.

The manual recovery was straightforward and safe, but it depended on out-of-band knowledge:

- stale merge artifacts can keep doctor/claim-gate posture degraded even when DB and canonical JSONL are in sync;
- the correct fix is to snapshot and move artifacts aside, not rebuild or overwrite the canonical store;
- the proof of success is a clean doctor plus clean sync status.

Beads should encode that workflow directly.

## Suggested tests

1. Fixture: valid `.beads/issues.jsonl`, valid `.beads/beads.db`, `dirty_count=0`, `jsonl_newer=false`, `db_newer=false`, plus stale `.beads/beads.left.jsonl` and `.beads/beads.base.jsonl`.
   - Expected: `doctor` reports `jsonl.merge_artifacts.status=warn`, includes file list, and marks writes/claim authority unsafe or recoverable.
2. Same fixture through `br repair merge-artifacts --snapshot --quarantine --json`.
   - Expected: artifacts are moved into a timestamped recovery directory, not deleted; postcheck reports doctor/sync clean.
3. Fixture with conflict markers in canonical `issues.jsonl` or artifacts.
   - Expected: repair refuses automatic quarantine and emits an inspect-first recovery action.
4. Fixture where DB and JSONL both have unsynced changes.
   - Expected: repair refuses to pick a winner and points to explicit DB-vs-JSONL merge workflow.
5. Robot contract test proving `br sync --status --json` or a shared authority field surfaces `merge_artifacts_present` so agents do not have to reconcile `doctor` separately.

**Comment by Dicklesworthstone** [comment]
Closing — the requested workflow exists and was verified end-to-end, and the discoverability gap you actually hit is now fixed in e478d7b9.

What already existed (verified empirically: planted a `beads.left.jsonl`, ran the repair, byte-checked the results):
- `br doctor --repair` quarantines stale merge artifacts via reversible renames into `.doctor/runs/<run-id>/quarantine/.beads/` — never deletes — and `br doctor undo <run-id>` reverses byte-deterministically. The JSON response includes `repaired: true` plus a `post_repair` block re-running ok/workspace_health/reliability_audit (the postcheck you asked for).
- `beads.base.jsonl` itself is no longer flagged at all — it's the protected 3-way merge anchor, excluded from both detector and fixer (since c34a12c9; your report was from br 0.1.13).

What was missing and is now in e478d7b9: the `jsonl.merge_artifacts` warning's details now classify each artifact (`artifact_kind`, conflict-marker scan) and advertise the structured recovery actions (`recovery: [{quarantine: "br doctor --repair"}, {undo: "br doctor undo <run-id>"}]`) plus the canonical JSONL path — so an agent seeing the warning knows the safe path without spelunking docs.
